src/zkp/pimod.rs

```diff
@@ -584,6 +584,38 @@ mod tests {
         }
     }
 
+    #[test]
+    fn modulus_factors_cannot_be_one() -> Result<()> {
+        let mut rng = init_testing();
+        let (_, _, q) = DecryptionKey::new(&mut rng).unwrap();
+        let one: BigNumber = BigNumber::from(1);
+        let modulus = one.clone() * q.clone();
+        let input = CommonInput { modulus };
+        let secret = ProverSecret {
+            p: one.clone(),
+            q: q.clone(),
+        };
+        let proof = PiModProof::prove(&input, &secret, &(), &mut transcript(), &mut rng);
+        assert!(proof.is_err());
+        Ok(())
+    }
+
+    #[test]
+    fn modulus_factors_cannot_be_composite() -> Result<()> {
```

Comment:
todo: add another check here that switches the order of `p` and `q` in the secret.

nit: use references to multiply for the modulus (`&random_odd_composite * &q`) to avoid cloning. You also shouldn't need the clone to make `secret` (but you will need it once when you add the check above).